src/marshmallow_annotations/builder.py

```diff
@@ -0,0 +1,236 @@
+from datetime import date, datetime, time, timedelta
+from decimal import Decimal
+from functools import wraps
+from types import MappingProxyType
+from typing import GenericMeta, List, Union, get_type_hints
+from uuid import UUID
+
+from marshmallow import Schema, fields
+from marshmallow.base import SchemaABC
+from marshmallow.class_registry import get_class as get_scheme
+from marshmallow.exceptions import MarshmallowError
+from marshmallow.schema import SchemaMeta
+
+NoneType = type(None)
+_UNION_TYPE = type(Union)
+
+
+class MarshmallowAnnotationError(MarshmallowError):
+    pass
+
+
+class AnnotationConversionError(MarshmallowAnnotationError):
+    pass
+
+
+def default_field_constructor(field):
+
+    def _(converter, subtypes, k):
+        return field(**k)
+
+    _.__name__ = f"{field.__name__}FieldConstructor"
+    return _
+
+
+def default_scheme_constructor(scheme_name):
+
+    def _(converter, subtypes, k):
+        return fields.Nested(scheme_name, **k)
+
+    _.__name__ = f"{scheme_name}FieldConstructor"
+    return _
+
+
+def _list_converter(converter, subtypes, k):
+    return fields.List(converter(subtypes[0]), **k)
+
+
+class TypeRegistry:
+    _registry = {
+        k: default_field_constructor(v)
+        for k, v in {
+            bool: fields.Boolean,
+            date: fields.Date,
+            datetime: fields.DateTime,
+            Decimal: fields.Decimal,
+            float: fields.Float,
+            int: fields.Integer,
+            str: fields.String,
+            time: fields.Time,
+            timedelta: fields.TimeDelta,
+            UUID: fields.UUID
```

Comment:
hm, what other fields should go in here

Reply:
tuple -> fields.List?
 mapping typing.Dict to fields.Dict probably isn't great but regular dict or mapping to fields.Dict could be worthwhile.

Reply:
still not sure. :\ :| :/